layer/layer_store.go

```diff
@@ -316,8 +316,7 @@ func (ls *layerStore) registerWithDescriptor(ts io.Reader, parent ChainID, descr
 	defer ls.layerL.Unlock()
 
 	if existingLayer := ls.getWithoutLock(layer.chainID); existingLayer != nil {
-		// Set error for cleanup, but do not return the error
-		err = errors.New("layer already exists")
+		logrus.Warnf("Layer %q already exists", layer.chainID)
```

Comment:
I am OK updating the comment if it is not clear. The way `err` is scoped in this function can be kind of confusing so having an extra statement about where it will get handled is valuable.